android/views/src/main/kotlin/com/boswelja/ephemeris/views/pager/InfiniteAnimatingPager.kt

```diff
@@ -91,13 +93,43 @@ internal class PageChangeAnimator(
     ): Boolean {
         val fromHeight = preInfo.bottom - preInfo.top
         val toHeight = postInfo.bottom - preInfo.top
-        heightAnimator.setIntValues(fromHeight, toHeight)
-        heightAnimator.addUpdateListener { animator ->
-            newHolder.itemView.layoutParams = newHolder.itemView.layoutParams.also {
-                it.height = animator.animatedValue as Int
+        if (fromHeight != toHeight) {
+            heightAnimator.setIntValues(fromHeight, toHeight)
+            heightAnimator.addUpdateListener { animator ->
+                newHolder.itemView.layoutParams = newHolder.itemView.layoutParams.also {
+                    it.height = animator.animatedValue as Int
+                }
             }
+            heightAnimator.addListener(
+                object : AnimatorListener {
```

Comment:
I wonder if there exists an AnimatorListener which acts like the core-ktx 
```
public inline fun TextView.doOnTextChanged(
    crossinline action: (
        text: CharSequence?,
        start: Int,
        before: Int,
        count: Int
    ) -> Unit
): TextWatcher = addTextChangedListener(onTextChanged = action)
```
So you only need to implement doOnAnimationEnd and doOnAnimationCancel so we can omit the no-ops 🤔

Reply:
There is not sadly, we'd need to include core-ktx for that. All for cleaning it up if you've got any ideas, it's kinda on the larger side

Reply:
https://developer.android.com/kotlin/ktx/extensions-list#for_androidanimationanimator damn there already is a ktx extension for it, they beat me to it :( my only suggestion to clean it up would be that

Reply:
I'm allergic to adding extra deps, maybe we can create one though

Reply:
hah, yeah can do that as a special case for us